internal/store/endpoint_test.go

```diff
@@ -167,9 +179,13 @@ func TestEndpointStoreWithLabels(t *testing.T) {
 				},
 			},
 			Want: metadata + `
-				kube_endpoint_address_available{endpoint="test-endpoint",namespace="default"} 6
-				kube_endpoint_address_not_ready{endpoint="test-endpoint",namespace="default"} 6
+				kube_endpoint_address_available_total{endpoint="test-endpoint",namespace="default"} 6
+				kube_endpoint_address_not_ready_total{endpoint="test-endpoint",namespace="default"} 6
 				kube_endpoint_annotations{endpoint="test-endpoint",annotation_app="foobar",namespace="default"} 1
+				kube_endpoint_address_available{endpoint="test-endpoint",namespace="default",addresses="127.0.0.1|10.0.0.1",ports="8080|8081"} 1 
+				kube_endpoint_address_available{endpoint="test-endpoint",namespace="default",addresses="172.22.23.202",ports="8443|9090"} 1
+				kube_endpoint_address_not_ready{endpoint="test-endpoint",namespace="default",addresses="192.168.1.1",ports="1234|5678"} 1
+				kube_endpoint_address_not_ready{endpoint="test-endpoint",namespace="default",addresses="192.168.1.3|192.168.2.2",ports="1234|5678"} 1
```

Comment:
This makes it hard to do queries in promql if they are all part of one list, do you really need the address as well? Having endpoint name should be enough for alerting or dashboard purposes. Afterwards you can always get the endpoint to further debug, no?